swarm/runtime/statsdb/ingestion.py

```
@@ -552,25 +552,73 @@ def ingest_events(self, events: List[Dict[str, Any]], run_id: str) -> int:
         # Set ingestion context to allow record_* calls
         _ingestion_context.active = True
         try:
-            return self._ingest_events_internal(events, run_id)
+            with self._lock:
+                self.connection.execute("BEGIN TRANSACTION")
+                try:
+                    count = self._ingest_events_internal(events, run_id)
+                    self.connection.execute("COMMIT")
+                    return count
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)